app/serializers/api/public/v1/serializable_course.rb

```diff
@@ -111,6 +112,14 @@ def enrichment_attribute(name, enrichment_name = name)
           @object.description
         end
 
+        attribute :provider_code do
+          @object.provider.provider_code
+        end
+
+        attribute :provider_type do
+          @object.provider.provider_type
+        end
```

Comment:
No this cannot be here, we need to conform with https://jsonapi.org/format/#fetching-sparse-fieldsets to properly utilise it, 

The controller should be updated to something like: 

```ruby

def index
          render jsonapi: paginate(courses),
                 include: include_param,
                 meta: { count: courses.count("course.id") },
                 class: API::Public::V1::SerializerService.call,
                 fields: fields
        end
        def fields
          { providers: provider_fields } if provider_fields.present?
        end

        def provider_fields
          params.dig(:fields, :providers)&.split(",")
        end

```
So that you can use a URL like 
http://localhost:3001/api/public/v1/recruitment_cycles/2021/courses?&fields[providers]=provider_type,code&include=provider

See 

![image](https://user-images.githubusercontent.com/470137/104316594-a0a68680-54d4-11eb-9364-faba3e80fc39.png)

Reply:
@defong - I can understand the rationale for not exposing provider attributes on the Course model (we were simply replicating v3 behaviour). I'm not clear though whether introducing sparse fields for an associated object is more beneficial than calling `course.provider.code?` ... am I missing something?

Reply:
We'll remove the `provider_code` and `provider_type` attributes